src/Couchbase.Lite.Shared/API/Sync/ReplicatorConfiguration.cs

```diff
@@ -211,49 +207,94 @@ public ReplicatorType ReplicatorType
 
         /// <summary>
         /// Gets or sets the replicator heartbeat keep-alive interval. 
-        /// The default Heartbeat is <c>5</c> min.
+        /// The default is null (5 min interval is applied). 
+        /// * <c>5</c> min interval is applied when Heartbeat is set to null.
+        /// * null will be returned when default <c>5</c> min interval is applied.
         /// </summary>
         /// <exception cref="ArgumentException"> 
         /// Throw if set the Heartbeat to less or equal to 0 full seconds.
         /// </exception>
-        public TimeSpan Heartbeat
+        public TimeSpan? Heartbeat
         {
-            get => Options.Heartbeat;
-            set => _freezer.PerformAction(() => Options.Heartbeat = value);
+            get 
+            {
+                if (Options.Heartbeat < 0)
+                    return null;
+                else
+                    return TimeSpan.FromSeconds(Options.Heartbeat); 
+            }
+
+            set
+            {
+                if (value != null) {
```

Comment:
Do nothing is default null value is set... But actually 5 min interval is applied in LiteCore...